### ISSUE TITLE
About Joint Training

### ISSUE THREAD

**Issue by DongfeiJi** [Canjie-Luo/Text-Image-Augmentation#10]
Thanks for your work, but I am wondering how to joint training, which is mentioned in the paper (Algorithm 1 Joint Learning Scheme).

**Comment by yanfjz** [comment]
I have two related questions as well:
1. Is the "moving state" S the deltas between src_pts and dst_pts `(src_pts - dst_pts)` which contains `2(N+1)` points?
2. Is the loss function for joint learning cross-entropy loss? If so, what is the representation of GT for this loss function?
Any explanation/idea/suggestion/discussion is welcomed :)

**Comment by Canjie-Luo** [comment]
@DongfeiJi 
This version is the default random augmentation. We guess it is sufficient for practical use.

**Comment by Canjie-Luo** [comment]
@yanfjz 
1. The "moving state" denotes the moving directions.
2. Yes, we use cross-entropy loss. The GT is the moving state that increases difficulty.

**Comment by zdz1997** [comment]
> @DongfeiJi
> This version is the default random augmentation. We guess it is sufficient for practical use.

I read the paper Learn to Augment: Joint Data Augmentation and Network Optimization
for Text Recognition.  
and  I'm so interested in joint training (Algorithm 1 Joint Learning Scheme)，can I ask for the whole code of paper?There is something I can't understand and I want to retrain it  .I'll  appreciate it very much .Here's my email zdz1954@126.com

**Comment by zdz1997** [comment]
I read the paper Learn to Augment: Joint Data Augmentation and Network Optimization
for Text Recognition.  
and  I'm so interested in joint training (Algorithm 1 Joint Learning Scheme)，can I ask for the whole code of paper?There is something I can't understand and I want to retrain it  .I'll  appreciate it very much .Here's my email zdz1954@126.com

**Comment by zdz1997** [comment]
@Canjie-Luo 
I read the paper Learn to Augment: Joint Data Augmentation and Network Optimization
for Text Recognition.  
and  I'm so interested in joint training (Algorithm 1 Joint Learning Scheme)，can I ask for the whole code of paper?There is something I can't understand and I want to retrain it  .I'll  appreciate it very much .Here's my email zdz1954@126.com

**Comment by DaeHwanGi** [comment]
@Canjie-Luo 

I have some questions about the paper.

1. Output size of the agent network is 2x(N+1)x2x2. Isn't just two coordinates enough to predict the direction of each point?
2. In Algorithm 1, randomly select one point in S and switch to the opposite direction to make S prime. Does this mean choosing one moving state in a mini-batch?

**Comment by shubham303** [comment]
I read the paper Learn to Augment: Joint Data Augmentation and Network Optimization
for Text Recognition.
and I'm so interested in joint training (Algorithm 1 Joint Learning Scheme)，can I ask for the whole code of paper?
my email: cs20m064@smail.iitm.ac.in

**Comment by firatkizilirmakk** [comment]
@shubham303 
Have you got the code?
 
I would be glad to get joint training code as well.
my email: firatkizilirmak34@gmail.com

Thanks in advance!

**Comment by matiascoronados** [comment]
@shubham303  
I would really appreciate if you can share that code with me!

my email: matias.coronado@usach.cl

Thanks :)

**Comment by shubham303** [comment]
@matiascoronados I don't have the code.

**Comment by Canjie-Luo** [comment]
Hey guys, due to the intellectual property protocol, I cannot release the code.
